Explain scoring

iankronquist opened the pull request:
One of the alternate strategies in issue #27. @willkg r?

iankronquist committed:
Add an explain method to the classifiers

iankronquist committed:
Add an explain method to the classifiers

iankronquist committed:
Add a test for the explain function

iankronquist committed:
Add a test for the explain function

iankronquist committed:
Add an explain function

iankronquist committed:
Add an explain function

iankronquist committed:
Tests for classifiers' explain functions

iankronquist committed:
Tests for classifiers' explain functions

iankronquist committed:
Docstrings!

iankronquist committed:
Docstrings!

willkg commented on spicedham/digitdestroyer.py:
```diff
@@ -26,7 +26,17 @@ def classify(self, response):
         If the responses consists entirely of numbers, return the filter_match
         value from the config file. Otherwise return filter_miss.
         """
+        return self.explain(response)[0]
```
Looks like this function is the same in all the subclasses. If that's true, you can put it in the superclass and nix all the extra code in the subclasses.

willkg commented on spicedham/nonsensefilter.py:
```diff
@@ -29,9 +29,19 @@ def classify(self, response):
         list_in_dict = lambda x, y: not self.backend.get_key(x, y, False)
         if all(imap(list_in_dict, repeat(classifier), response)):
-            return self.filter_match
+            return self.filter_match, 'None of the response\'s words were\
```
I'd do this as:

```
return (self.filter_match,
        'None of the response\'s words were found in the backend')
```

What you're doing uses `\` at the end of the line as a continuation for the string and ... that's kind of a gross thing to do plus it has problems if there's accidentally extra space after the `\` which is hard to see.

willkg commented:
I had a couple of hygiene issues, but otherwise this looks good.

iankronquist committed:
Merge pull request #33 from iankronquist/docs

Docs

iankronquist committed:
Remove classify functions

They've all been bumped to the superclass.
Also, split tuple onto different lines, not strings.

iankronquist committed:
Add an explain function, implement classify

iankronquist committed:
Merge branch 'explain-scoring' of github.com:iankronquist/spicedham into explain-scoring

Conflicts:
	spicedham/nonsensefilter.py
